client/src/components/CompaniesAutoComplete.tsx

```diff
@@ -155,7 +155,7 @@ export default class CompaniesAutoComplete extends React.Component<CompaniesAuto
               <Button onClick={this.handleClose} color="primary">
                 Cancel
               </Button>
-              <Button type="submit" color="primary">
+              <Button type="submit" color="primary" disabled={this.state.dialogValue.name === ""}>
```

Comment:
I believe that this line disables the submit button if nothing is in the company field. This prevents a bug we have from occurring.